src_c/polygon.c

```diff
@@ -305,6 +305,129 @@ pg_polygon_get_vertices(pgPolygonObject *self, void *closure)
     return _pg_polygon_vertices_aslist(&self->polygon);
 }
 
+static int
+pg_polygon_ass_vertex(pgPolygonObject *self, Py_ssize_t i, PyObject *v)
+{
+    pgPolygonBase *poly = &self->polygon;
+
+    if (i < 0) {
+        i += poly->verts_num;
+    }
+
+    if (i < 0 || i > poly->verts_num) {
+        PyErr_SetString(PyExc_IndexError, "Invalid vertex Index");
+        return -1;
+    }
+
+    double v1, v2;
+    if (!pg_TwoDoublesFromObj(v, &v1, &v2)) {
+        PyErr_SetString(PyExc_TypeError, "Must assign numeric values");
+        return -1;
+    }
+    poly->vertices[i * 2] = v1;
+    poly->vertices[i * 2 + 1] = v2;
+    return 0;
+}
+
+static Py_ssize_t
+pg_polygon_seq_length(pgPolygonObject *self)
+{
+    pgPolygonBase *poly = &self->polygon;
+    return poly->verts_num;
```

Comment:
use `pgPolygon_GETVERTSNUM`
```suggestion
    return pgPolygon_GETVERTSNUM(self);
```